### PR TITLE
Handle error when pass wrong uri to SvgFromUri component

### LINKED ISSUE #2148: Bad svg uri content is causing app to freeze
# Bug

image

#### Unexpected behavior

Happens on all scenarios

## Environment info

"react-native-svg": "^13.14.0",
"react-native": "0.72.5",
"react": "18.2.0",
"expo": "^49.0.0",

## Steps To Reproduce

Load https://only1.io/like-token.svg using SvgFromUri, I was expecting to be a svg file but looks like the file doesn't exist anymore and the whole page is rendered.

Describe what you expected to happen:

1. If the file doesn't exist just throw the error and that's it but the app freezes, the buttons do not respond anymore
2. App should continue working even if the error was thrown

## Short, Self Contained, Correct (Compilable), Example

NA

### PR DESCRIPTION
**Pull request by bohdanprog**
# Summary

We want to add the ability to handle the Error when someone passes the wrong URI to the `SvgFromUri` component.

Closes #2148 

## Test Plan

Manual tests in TestsExample app. (Test2148)

### CODE REVIEW

File: src/xml.tsx [hunk 1]
```diff
@@ -171,10 +171,15 @@ export class SvgFromXml extends Component<XmlProps, XmlState> {
   }
 
   parse(xml: string | null) {
+    const { onError = err } = this.props;
     try {
       this.setState({ ast: xml ? parse(xml) : null });
     } catch (e) {
```
Comment (WoLewicki) [inline src/xml.tsx:177]: Maybe we could do `e: Error` here to not cast below?

File: src/xml.tsx [hunk 2]
```diff
@@ -171,10 +171,15 @@ export class SvgFromXml extends Component<XmlProps, XmlState> {
...
-      console.error(e);
+      const error = e as Error;
+      onError({
+        ...error,
+        message: `Couldn't parse SVG, case: ${error.message}`,
```
Comment (WoLewicki) [inline src/xml.tsx:181]:
```suggestion
        message: `[RNSVG] Couldn't parse SVG, reason: ${error.message}`,
```

Would be better maybe?

File: TestsExample/App.js [hunk 3]
```diff
@@ -14,7 +14,8 @@ import Test2196 from './src/Test2196';
 import Test2266 from './src/Test2266';
 import Test1986 from './src/Test1986';
 import Test2276 from './src/Test2276';
+import Test2148 from './src/Test2148';
 
 export default function App() {
-  return <ColorTest />;
+  return <Test2148 />;
```
Comment (WoLewicki) [inline TestsExample/App.js:20]: I think we want to leave `ColorTest` as the default one. Also, could you sort the imports so its easier to go through them?

**Review by WoLewicki — COMMENTED** [review]
Please answer the comments, other than that LGTM :tada:

File: TestsExample/App.js (hunk 3 above)
Comment (bohdanprog) [reply TestsExample/App.js:20]: Okay, I'm gonna change that :)

File: src/xml.tsx (hunk 2 above)
Comment (bohdanprog) [reply src/xml.tsx:181]: Yep, it looks better, thanks 👍

File: src/xml.tsx (hunk 1 above)
Comment (bohdanprog) [reply src/xml.tsx:177]: As far as I know, we can't catch error types, or is there something I'm missing?

Comment (WoLewicki) [reply src/xml.tsx:177]: Oh, didn't know that. Then it's fine if we are sure that the `e` is of type `Error`.

**Review by WoLewicki — APPROVED** [review]
LGTM!

**Pull request merged**